lib/pipette/handler.ex

```
@@ -8,9 +8,10 @@ defmodule Pipette.Handler do
   @callback call(value :: any, args :: list(any), ip :: IP.t()) :: return_t
   @optional_callbacks call: 0, call: 1, call: 2, call: 3
 
-  def handle(handler, %IP{} = ip) do
+  def handle(handler, %IP{ref: ref} = ip) do
     case perform(handler, ip) do
-      %IP{} = ip -> ip
+      %IP{ref: ^ref} = ip -> ip
+      %IP{} -> IP.update(ip, {:error, "IP.ref mismatch"})
```

I think, if we go down that path to "harden" or "strictly enforce" rules on `IP` in the handler module, then we should raise an error here, to fail the complete thing.

These errors will happen exclusively during development, so the developer should be warned as soon as possible.

An error route is hiding it in some output down the chain, and it is harder to reason about why this is enforced this way.